### ISSUE TITLE
Room history settings don't work

### ISSUE THREAD

**Issue by alexgleason** [vector-im/riot-web#5247]
I invited someone to a room. The room history was set to "members only (since they were invited)." It didn't work. The person joined the room and was able to scroll up and see the history.

This was incredibly awkward and extremely bad! Please fix this!

**Comment by turt2live** [comment]
I can confirm this.

My view:
![image](https://user-images.githubusercontent.com/1190097/31464232-1acf9fe6-ae8e-11e7-9002-b7481c59318d.png)

`@temp:t2l.io`'s view
![image](https://user-images.githubusercontent.com/1190097/31464214-0c9e83ce-ae8e-11e7-9b56-3af7956bbf7f.png)

**Comment by lukebarnard1** [comment]
Ok, so @erikjohnston has explained that you can only change the history visibility for messages sent _after_ the history visibility change. Riot needs to spell this out much more clearly I think.

**Comment by lampholder** [comment]
I post this only to highlight how _insufficient_ a pop up is for communicating this very important message, since I think what we have here is (yet more) evidence that popups are essentially "invisible".

Anyway, when you make this change Riot pops up a warning explaining exactly what's going to happen:
![image](https://user-images.githubusercontent.com/1922197/31493150-df837a08-af45-11e7-9c20-a777819cc7d3.png)

**Comment by turt2live** [comment]
To be honest, I didn't even read the dialog in my testing. I just assumed it wasn't a problem, and was surprised when the bug could be reproduced. 

Some nice bold/red text explaining it would probably catch the eye enough to reconsider skimming the dialog.

**Comment by marykatefain** [comment]
I don't think that popup exists at all on mobile. I just tested it myself because I swear I never saw that before... I use riot almost exclusively on mobile.

**Comment by alexgleason** [comment]
I changed the history visibility on Android which did not display this popup message.

For what it's worth, I've seen the message on desktop and even then misunderstood what it meant.

**Comment by lampholder** [comment]
Yeah - popups are awful, _esp_ at conveying important information like this.

So let's leave this bug scoped to doing something better in Riot web - I'll make another bug for android.

**Comment by lampholder** [comment]
https://github.com/vector-im/riot-android/issues/1703

**Comment by ara4n** [comment]
The root cause of this is https://github.com/matrix-org/matrix-doc/issues/533
